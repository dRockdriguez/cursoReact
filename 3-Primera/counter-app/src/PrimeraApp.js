import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, subSaludo }) => {
  return (
    <>
      <h1>{ saludo }</h1>
      <p>{ subSaludo }</p>
    </>
  );
};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subSaludo: PropTypes.string
};

PrimeraApp.defaultProps = {
    subSaludo: "Esto es el subSaludo"
};

export default PrimeraApp;
