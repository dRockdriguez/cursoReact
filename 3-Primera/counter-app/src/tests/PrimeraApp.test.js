import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import PrimeraApp from "../PrimeraApp";
import React from "react";
import { shallow } from "enzyme";

describe("Pruebas de <PrimeraApp />", () => {
  /*test("debe mostrar el saludo 'Hola!'", () => {
    const saludo = "Hola!";

    const { getByText } = render(<PrimeraApp saludo={saludo} />);

    expect(getByText(saludo)).toBeInTheDocument();
  });*/

  test("debe mostrar el saludo correctamente con enzyme", () => {
    const saludo = "Hola!";

    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el subtitulo por defecto correctamente", () => {
    const saludo = "Hola!";
    const subtitulo = "Esto es el subSaludo";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    const p = wrapper.find("p").text();

    expect(p).toBe(subtitulo)
  });
});
