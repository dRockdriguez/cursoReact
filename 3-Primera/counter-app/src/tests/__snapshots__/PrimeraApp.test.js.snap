// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <PrimeraApp /> debe mostrar el saludo correctamente con enzyme 1`] = `
<Fragment>
  <h1>
    Hola!
  </h1>
  <p>
    Esto es el subSaludo
  </p>
</Fragment>
`;
